rfcs/RFC: Handling XLA Exceptions in Python

```diff
@@ -0,0 +1,101 @@
+RFC: Handling XLA Exceptions in Python
+
+Status	Propose
+Author	Vinay Kumar Burugu
+Collaborator	Loki Ravi
+PR	https://github.com/tensorflow/community/pull/435
+
+Objective
+
+The aim of this RFC is to:
+
+1. Any exception/status code thrown by XLA to be uniquely named (independent of TensorFlow).
+2. XLA exceptions to be readily available to front-end frameworks so that they can be exposed to users.
+
+Motivation
+
+1. Though XLA was developed as a part of Tensorflow, it is also used by PyTorch, JAX, etc. There is a need for exceptions thrown by XLA to be named independent of Tensorflow. 
+2. Today, XLA defines XlaRuntimeError in the xla client. However, this exception is not readily available to front-ends. Refer to this (https://github.com/google/jax/pull/10676) effort in JAX to expose this Exception to users.
+
+User Benefit
+
+1. *Simplified contract with front-ends*: Add a single line of code to front-ends to import XLA exceptions. For eg: from import xla.exceptions as xla_exceptions.
+2. *Better exception names in XLA*: Derive readable exceptions like XlaInvalidArgumentError from XlaError to replace the existing ValueError. Additionally, this would make it easier to catch/trap Exceptions from XLA and attribute errors to XLA.
+
+Design Proposal
+
+This proposal would replace existing std:: and py:: exceptions thrown by the CPP layers with new custom exceptions which will all be extended from XlaError base class. All status objects originating from XLA will be replaced by XLA specific status codes. Equivalent python exceptions will be defined under a common module xla.exceptions which will contain translations for status objects (previously TensorFlow exceptions (https://github.com/tensorflow/tensorflow/blob/87462bfac761435a46641ff2f10ad0b6e5414a4b/tensorflow/python/framework/errors_impl.py)) as well as direct throws from XLA (previously std library exceptions).
```

Comment:
Can you clarify what you mean by `All status objects originating from XLA will be replaced by XLA specific status codes`? Do you want to expose the literals in a separate, XLA specific python module or modify them? Do you want a single exception that covers all status codes or sub-exceptions for each?

Reply:
The Status Object updates the error codes from [errorcodes.proto](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tsl/protobuf/error_codes.proto) which are common to both XLA and tensorflow. I wanted to use unique codes for XLA and create corresponding python Exceptions for those error codes so that we see unique XLA exceptions corresponding to those new error codes added. All this effort is being proposed just to identify whether the root cause is framework or XLA instantly based on the Python Error / Error code etc.

Reply:
If the error codes get wrapped into an 'XlaError' exception at the XLA API boundary, why would we need different error numbers? I agree we should expose the numbers from XLA, as well, but it is not clear to me why they could not be the same numbers?